lib/cloudware/models/concerns/provider_client.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+module Cloudware
+  module Models
+    module Concerns
+      module ProviderClient
+        extend Memoist
+
+        delegate :provider, to: Config
+        delegate :region, to: :context
+
+        private
+
+        def provider_client
+          if provider == 'aws'
+            require 'providers/AWS'
```

Comment:
Why are the provider source code files capitalized?  That's a very unusual thing to do for ruby source code files.

Reply:
This is to avoid conflicts with the provider `sdk` which are refer to as `Aws` and `Azure`. In the case of `AWS` this is okay as it is an acronym which does have a precedence (e.g. `PP`)

Reply:
Resolved in https://github.com/alces-software/cloudware/commit/b0289a148dc933224090bf18abb158553662ab6e